routing/endpointregistry.go

```diff
@@ -164,15 +166,19 @@ func (r *EndpointRegistry) updateStats() {
 			e.totalFailedRoundTrips[nextSlot].Store(0)
 			e.totalRequests[nextSlot].Store(0)
 
+			newProbabilityStored := false
 			failed := e.totalFailedRoundTrips[curSlot].Load()
 			requests := e.totalRequests[curSlot].Load()
 			if requests > r.minRequests {
 				failedRoundTripsRatio := float64(failed) / float64(requests)
-				if failedRoundTripsRatio > 0.0 {
+				if failedRoundTripsRatio > r.minHealthCheckDropProbability {
 					log.Infof("Passive health check: marking %q as unhealthy due to failed round trips ratio: %0.2f", key, failedRoundTripsRatio)
+					e.healthCheckDropProbability.Store(min(failedRoundTripsRatio, r.maxHealthCheckDropProbability))
+					newProbabilityStored = true
 				}
-				e.healthCheckDropProbability.Store(min(failedRoundTripsRatio, r.maxHealthCheckDropProbability))
-			} else {
+			}
+
+			if !newProbabilityStored {
 				e.healthCheckDropProbability.Store(0.0)
```

Comment:
Shouldn't we store minHealthCheckDropProbability instead of zero here?

Reply:
Ah, ok, we should change it when it is above min. But why do we changed else branch instead of just changing `if failedRoundTripsRatio > 0.0`?

Reply:
I think this way would be the most convenient. This way we will get rid of else branch at all.

```
newDropProbability := 0.0 // default value
if manyRequests {
  if failedRatioIsBig {
    newDropProbability = failedRatio //non-default value
  }
}
e.healthCheckDropProbability.Store(newDropProbability)
```